File: boj/14719/main.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int w, h;
int v[501];

void input() {
  cin >> h >> w;
  for (int i = 0; i < w; i++) {
    cin >> v[i];
  }
}

int main() {
  cin.tie(nullptr);
  ios_base::sync_with_stdio(false);

  input();
  int ans = 0;
  for (int i = 1; i < w - 1; i++) {
    int left = *max_element(v, v + i);
    int right = *max_element(v + i, v + w);
    int x = min(left, right) - v[i];
    ans += max(x, 0);
  }

  cout << ans << '\n';
  return 0;
}

File: boj/14719/main.rs
use std::io::{self, BufRead};
use std::cmp::min;

fn main() {
    let stdin = io::stdin();
    let v: Vec<String> = stdin.lock().lines().skip(1)
        .map(|x| x.unwrap()).collect();
    let v: Vec<i32> = v[0]
        .split_whitespace()
        .map(|x| x.parse().unwrap())
        .collect();

    let mut ans = 0;
    let n = v.len();

    for i in 1..n-1 {
        let left = &v[0..i].iter().max().unwrap();
        let right = &v[i..n].iter().max().unwrap();
        let x = *min(left, right) - v[i];
        ans += if x > 0 { x } else { 0 };
    }
    println!("{}", ans);
}
